bob/src/cluster/quorum.rs

```diff
@@ -239,6 +246,51 @@ impl Quorum {
             Ok(())
         }
     }
+
+    pub(crate) async fn delete_aliens(
+        &self,
+        mut failed_nodes: Vec<String>,
+        key: BobKey,
+    ) -> Result<(), Error> {
+        if failed_nodes.is_empty() {
+            warn!(
+                "DELETE[{}] trying to aliens, but there are no failed nodes: unreachable",
+                key
+            );
+            return Err(Error::internal());
+        }
+        trace!("selection of free nodes available for data writing");
+        let sup_nodes = self.mapper.get_support_nodes(key, failed_nodes.len());
+        debug!("DELETE[{}] sup delete nodes: {:?}", key, &sup_nodes);
+        let nodes_need_remote_backup: Vec<_> = failed_nodes.drain(..sup_nodes.len()).collect();
+        let queries: Vec<_> = sup_nodes
+            .into_iter()
+            .zip(nodes_need_remote_backup)
+            .map(|(node, remote_node)| (node, DeleteOptions::new_alien(vec![remote_node])))
+            .collect();
+        debug!("DELETE[{}] additional alien requests: {:?}", key, queries);
+        if let Err(sup_nodes_errors) = delete_sup_nodes(key, &queries).await {
+            debug!("support nodes errors: {:?}", sup_nodes_errors);
+            failed_nodes.extend(
+                sup_nodes_errors
+                    .iter()
+                    .map(|err| err.node_name().to_owned()),
+            )
+        };
+        debug!("need additional local alien copies: {}", failed_nodes.len());
```

Comment:
Add `if` check that we need to write deletion records to local node aliens (`failed_nodes.len() > 0`)